pkg/network/msg_handler.go

```diff
@@ -10,21 +14,38 @@ const (
 
 // MsgHandler handle messages received from a service
 type MsgHandler struct {
-	logger logging.Logger
-	amqp   *Amqp
+	logger       logging.Logger
+	amqp         *Amqp
+	msgPublisher *MsgPublisher
 }
 
 func (mc *MsgHandler) onMsgReceived(msgChan chan InMsg) {
 	for {
 		msg := <-msgChan
 		mc.logger.Infof("Exchange: %s, routing key: %s", msg.Exchange, msg.RoutingKey)
 		mc.logger.Infof("Message received: %s", string(msg.Body))
+
+		switch msg.RoutingKey {
+		case "device.register":
+			msgParsed := RegisterRequestMsg{}
```

Comment:
Move this code to a function in order to improve the switch case readability.

Reply:
Done

Reply:
Done